iac-extensions/arm/src/test/java/org/sonar/iac/arm/checks/ManagedIdentityCheckTest.java

```diff
@@ -0,0 +1,234 @@
+      "Microsoft.SecurityAndCompliance/privateLinkServicesForMIPPolicySync",
+      "Microsoft.SecurityAndCompliance/privateLinkServicesForO365ManagementActivityAPI",
+      "Microsoft.SecurityAndCompliance/privateLinkServicesForSCCPowershell",
+      "Microsoft.ServiceBus/namespaces",
+      "Microsoft.ServiceFabric/clusters/applications",
+      "Microsoft.ServiceFabric/managedclusters/applications",
+      "Microsoft.SignalRService/signalR",
+      "Microsoft.SignalRService/webPubSub",
+      "Microsoft.Solutions/applianceDefinitions",
+      "Microsoft.Solutions/appliances",
+      "Microsoft.Solutions/applicationDefinitions",
+      "Microsoft.Solutions/applications",
+      "Microsoft.Sql/managedInstances",
+      "Microsoft.Sql/servers",
+      "Microsoft.Sql/servers/databases",
+      "Microsoft.SqlVirtualMachine/sqlVirtualMachines",
+      "Microsoft.Storage/storageAccounts",
+      "Microsoft.StorageCache/amlFilesystems",
+      "Microsoft.StorageCache/caches",
+      "Microsoft.StreamAnalytics/streamingjobs",
+      "Microsoft.Synapse/workspaces",
+      "Microsoft.VideoIndexer/accounts",
+      "Microsoft.VirtualMachineImages/imageTemplates",
+      "Microsoft.Web/sites",
+      "Microsoft.Web/sites/slots",
+      "Microsoft.Web/staticSites",
+      "Microsoft.Workloads/monitors",
+      "Microsoft.Workloads/monitors/providerInstances",
+      "Microsoft.Workloads/phpWorkloads",
+      "Microsoft.Workloads/sapVirtualInstances");
+  }
+
+  @MethodSource(value = "managedIdentityTypes")
+  @ParameterizedTest(name = "[{index}] JSON should check managed identity check for type {0}")
+  void shouldCheckManagedIdentityCheckJson(String type) {
+    String content = ArmTestUtils.readTemplateAndReplace("ManagedIdentityCheck/managedIdentityCheck_template.json", type);
+    verifyContent(content, check,
+      issue(8, 14, 8, 54, "Omitting the \"identity\" block disables Azure Managed Identities. Make sure it is safe here."),
+      issue(23, 16, 23, 22, "Make sure that disabling Azure Managed Identities is safe here."),
+      issue(38, 6, 40, 7, "Omitting the \"type\" in \"identity\" block disables Azure Managed Identities. Make sure it is safe here."));
+  }
+
+  @MethodSource(value = "managedIdentityTypes")
+  @ParameterizedTest(name = "[{index}] Bicep should check managed identity check for type {0}")
+  void shouldCheckManagedIdentityCheckBicep(String type) {
+    String content = ArmTestUtils.readTemplateAndReplace("ManagedIdentityCheck/managedIdentityCheck_template.bicep", type);
+
+    BicepVerifier.verifyContent(content, check,
+      issue(1, 9, 1, 39, "Omitting the \"identity\" block disables Azure Managed Identities. Make sure it is safe here."),
+      issue(18, 10, 18, 16, "Make sure that disabling Azure Managed Identities is safe here."),
+      issue(33, 2, 35, 3, "Omitting the \"type\" in \"identity\" block disables Azure Managed Identities. Make sure it is safe here."));
```

Comment:
Why not define the expected issues directly in the file as comment?
Easier to define and to read afterwards.

Reply:
Because it's for different types, I have to first read the content, replace `${type}` and then check issues. Other methods expect a filename as argument.